dulwich/refs.py

```diff
@@ -687,6 +688,14 @@ def set_if_equals(self, name, old_ref, new_ref, committer=None,
         except (KeyError, IndexError):
             realname = name
         filename = self.refpath(realname)
+
+        # make sure none of the ancestor folders is an existing ref
+        probe_ref = os.path.dirname(realname)
+        while probe_ref:
+            if self.read_ref(probe_ref) is not None:
```

Comment:
Please check just the packed refs here; for non-packed refs you'd already get a EISDIR error anyway. This is a heavily used code path, and this new behaviour is adding several extra stats every time a ref is changed.